### ISSUE TITLE
SMTP via TLS/SSL with SparkPost Fails

### ISSUE THREAD

**Issue by plunkettscott** [FusionAuth/fusionauth-issues#122]
### Description 
When selecting TLS/SSL for the SMTP connection and entering settings for the [SparkPost SMTP API](https://developers.sparkpost.com/api/smtp/#header-client-configuration), the connection is never made.

### Steps to reproduce
1. Go to SMTP Settings
2. Enter [SparkPost settings](https://developers.sparkpost.com/api/smtp/#header-client-configuration) and enable SSL or TLS
3. Complete an action that would send an email, such as registering a user.

### Expected behavior
The email is delivered via the SMTP API

### Actual behavior
The email is not devliered and no connection to the SMTP API happens

### Platform
- Kubernetes Deployment
- FusionAuth 1.5.0

### Additional context
Selecing NONE as the mail security option results in emails successfully being delievered.

**Comment by robotdan** [comment]
Are you using port `587` - and is there anything in the FusionAuth log?

**Comment by plunkettscott** [comment]
Yes, port 587. And nothing in the logs related to that, just typical info stuff.

**Comment by robotdan** [comment]
Want to try to enable `DEBUG` and see if anything interesting shows up in the log? 

This will return the current log level for the package:  (where api key is 'api-key')

```
curl "http://localhost:9011/api/logger?name=org.primeframework.email" -HAuthorization:api-key
```

Set the log level to `DEBUG` for the package:  (where api key is 'api-key')

```
curl "http://localhost:9011/api/logger?name=org.primeframework.email&level=debug" -HAuthorization:api-key
```

You'll of course need to modify the host and api key. :-)

**Comment by robotdan** [comment]
I may have recreated, I configured SparkPost and I stepped through and we're receiving the following exception: 

```
javax.mail.MessagingException: Could not convert socket to TLS;
  nested exception is:
	javax.net.ssl.SSLHandshakeException: Remote host closed connection during handshake
```

This should be enough for me to debug, so don't worry about enabling debug. I'll update this issue when I know more.

**Comment by robotdan** [comment]
I was able to get past this exception by providing some additional configuration to the SMTP configuration.  This is internal, and not user-modifiable. 

```
properties.put("mail.smtp.ssl.protocols", "TLSv1.2"); 
```

`mail.smtp.ssl.protocols`  as defined by https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html

I need to do a bit more research to see if this is unique to SparkPost or something we need to expose via the UI or something like that.

**Comment by robotdan** [comment]
Fixed in 1.6.0, and we'll also now have the option to override and add additional SMTP mail properties if we run into this type of thing in the future.
